src/components/common/Menu/hooks/useMenu.ts

```diff
@@ -1,22 +1,26 @@
 import { PopperProps as MPopperProps } from "@mui/material";
 import { MouseEvent, useCallback, useMemo, useState } from "react";
 
-export interface UseMenu {
-  anchorEl: MPopperProps["anchorEl"];
+export interface UseMenu<
+  E extends MPopperProps["anchorEl"] = MPopperProps["anchorEl"]
+> {
+  anchorEl: E | null;
   onClose: () => void;
   onDisableScrollLock: () => void;
   onEnableScrollLock: () => void;
-  onOpen: (event: MouseEvent<HTMLElement>) => void;
-  onToggleOpen: (event: MouseEvent<HTMLElement>) => void;
+  onOpen: (event: MouseEvent<E>) => void;
```

Comment:
Updated the useMenu hook to take a generic `E` to support use by both Mui Popper and Menu.